stl/inc/xutility

```diff
@@ -4352,10 +4352,11 @@ template <>
 struct _Is_character_or_byte_or_bool<bool> : true_type {};
 
 // _Fill_memset_is_safe determines if _FwdIt and _Ty are eligible for memset optimization in fill
+// Need to explicity test for volatile because _Unwrap_enum_t discards qualifiers.
 template <class _FwdIt, class _Ty, bool = is_pointer_v<_FwdIt>>
 _INLINE_VAR constexpr bool _Fill_memset_is_safe = conjunction_v<is_scalar<_Ty>,
     _Is_character_or_byte_or_bool<_Unwrap_enum_t<remove_reference_t<_Iter_ref_t<_FwdIt>>>>,

```

Comment:
I am wondering whether the correct fix would be to reject volatile for `_Unwrap_enum_t`

Reply:
I guess you could make it not discard volatile, but it's also used in `_Ptr_cat_helper`, which is indirectly used in many other places. You could also make a new template to have that behavior, but this seems simpler to me.

Reply:
My point is, would `volatile meow` be correct in any other optimization?

Reply:
I don't think so, `memset` doesn't accept a volatile pointer. My understanding of the conversation for #1183 was also that `memset` is never appropriate for volatile objects.